services/gitlab/gitlab-merge-requests.tester.js

```diff
@@ -0,0 +1,169 @@
+import Joi from 'joi'
+import { createServiceTester } from '../tester.js'
+import { isMetric, isMetricOpenIssues } from '../test-validators.js'
+
+export const t = await createServiceTester()
+
+t.create('Merge Requests (project not found)')
+  .get('/open/guoxudong.io/shields-test/do-not-exist.json')
+  .expectBadge({
+    label: 'merge requests',
+    message: 'project not found',
+  })
+
+/**
+ *  Opened issue number case
+ */
+t.create('Opened merge requests')
+  .get('/open/guoxudong.io/shields-test/issue-test.json')
+  .expectBadge({
+    label: 'merge requests',
+    message: isMetricOpenIssues,
+  })
+
+t.create('Open merge requests raw')
+  .get('/open-raw/guoxudong.io/shields-test/issue-test.json')
+  .expectBadge({
+    label: 'open merge requests',
+    message: isMetric,
+  })
+
+t.create('Open merge requests by label is > zero')
+  .get('/open/guoxudong.io/shields-test/issue-test.json?labels=discussion')
+  .expectBadge({
+    label: 'discussion merge requests',
+    message: isMetricOpenIssues,
+  })
+
+t.create('Open merge requests by  multi-word label is > zero')
+  .get(
+    '/open/guoxudong.io/shields-test/issue-test.json?labels=discussion,enhancement'
+  )
+  .expectBadge({
+    label: 'discussion,enhancement merge requests',
+    message: isMetricOpenIssues,
+  })
+
+t.create('Open merge requests by label (raw)')
+  .get('/open-raw/guoxudong.io/shields-test/issue-test.json?labels=discussion')
+  .expectBadge({
+    label: 'open discussion merge requests',
+    message: isMetric,
+  })
+
+t.create('Opened merge requests by Scoped labels')
+  .get('/open/gitlab-org%2Fgitlab.json?labels=test,failure::new')
+  .expectBadge({
+    label: 'test,failure::new merge requests',
+    message: Joi.string().regex(
+      /^([0-9]+[kMGTPEZY]?|[1-9]\.[1-9][kMGTPEZY]) open$/
+    ),
+  })
+
+/**
+ *  Closed issue number case
+ */
+t.create('Closed merge requests')
+  .get('/closed/guoxudong.io/shields-test/issue-test.json')
+  .expectBadge({
+    label: 'merge requests',
+    message: Joi.string().regex(
+      /^([0-9]+[kMGTPEZY]?|[1-9]\.[1-9][kMGTPEZY]) closed$/
+    ),
+  })
+
+t.create('Closed merge requests raw')
+  .get('/closed-raw/guoxudong.io/shields-test/issue-test.json')
+  .expectBadge({
+    label: 'closed merge requests',
+    message: isMetric,
+  })
+
+t.create('Closed merge requests by label is > zero')
+  .get('/closed/guoxudong.io/shields-test/issue-test.json?labels=bug')
+  .expectBadge({
+    label: 'bug merge requests',
+    message: Joi.string().regex(
+      /^([0-9]+[kMGTPEZY]?|[1-9]\.[1-9][kMGTPEZY]) closed$/
+    ),
```

Comment:
Same comment as above, and if necessary we can use the alternatives format:

```suggestion
    message: Joi.alternatives(isMetricClosedIssues, Joi.equal('0 closed')),
```